nopol/src/main/java/fr/inria/lille/commons/trace/SpecificationTestCasesListener.java

```diff
@@ -67,7 +67,8 @@ private void addToSpec(Map<String, Object> inputs, T output) {
 		}
 
 		if (output.equals(reference) && this.keys.equals(inputs.keySet())) {
-			this.logger.warn("You may have some redundant test: same input and same outcome, only one will be used: discarded.");// case 3
```

Comment:
you have redundant tests:  same input and same outcome, only one will be used to speed-up synthesis